Scripts/Runtime/Entities/ManagedReference/ManagedReferenceStore.cs

```diff
@@ -12,20 +12,27 @@ namespace Anvil.Unity.DOTS.Entities
     /// </summary>
     public static class ManagedReferenceStore
     {
-        private static readonly Logger s_Logger = Log.GetStaticLogger(typeof(ManagedReferenceStore));
-        private static readonly Dictionary<uint, IComponentReferencable> s_IDToInstance = new Dictionary<uint, IComponentReferencable>();
-        private static readonly Dictionary<IComponentReferencable, uint> s_InstanceToID = new Dictionary<IComponentReferencable, uint>();
+        private static readonly Logger s_Logger;
+        private static readonly Dictionary<uint, IComponentReferencable> s_IDToInstance;
+        private static readonly Dictionary<IComponentReferencable, uint> s_InstanceToID;
         private static IDProvider s_IDProvider;
 
-#if UNITY_EDITOR
+        static ManagedReferenceStore()
+        {
+            s_Logger = Log.GetStaticLogger(typeof(ManagedReferenceStore));
+            s_IDToInstance = new Dictionary<uint, IComponentReferencable>();
+            s_InstanceToID = new Dictionary<IComponentReferencable, uint>();
+
+            Init();
+        }
+
         [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
         private static void Init()
```

Comment:
Does this not get called twice in Editor? 

It may not matter for this function in particular but I've run into this problem with this pattern. You may want to wrap in a #if check for UNITY_EDITOR or it may just be fine to let it be called twice.

Reply:
You're right. That's risky to have IDs reset if something was triggering the static constructor before subsystem registration happened. (which was happening!)

I've removed the `Init` call from the constructor and changed to `RuntimeInitializeLoadType.AfterAssembliesLoaded` which works in builds and editor.